tests/Unit/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Test_Bjorhus.cpp

```diff
@@ -0,0 +1,193 @@
+    const tnsr::iaa<DataVector, Dim, frame>& phi,
+    // c.f. dg_interior_temporary_tags
+    const tnsr::I<DataVector, Dim, frame>& coords,
+    const Scalar<DataVector>& gamma1, const Scalar<DataVector>& gamma2,
+    const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& shift,
+    const tnsr::AA<DataVector, Dim, Frame::Inertial>& inverse_spacetime_metric,
+    const tnsr::A<DataVector, Dim, Frame::Inertial>&
+        spacetime_unit_normal_vector,
+    const tnsr::a<DataVector, Dim, Frame::Inertial>&
+        spacetime_unit_normal_one_form,
+    const tnsr::iaa<DataVector, Dim, Frame::Inertial>& three_index_constraint,
+    const tnsr::a<DataVector, Dim, frame>& gauge_source,
+    const tnsr::ab<DataVector, Dim, frame>& spacetime_deriv_gauge_source,
+    // c.f. dg_interior_dt_vars_tags
+    const tnsr::aa<DataVector, Dim, frame>& dt_spacetime_metric,
+    const tnsr::aa<DataVector, Dim, frame>& dt_pi,
+    const tnsr::iaa<DataVector, Dim, frame>& dt_phi,
+    // c.f. dg_interior_deriv_vars_tags
+    const tnsr::iaa<DataVector, Dim, frame>& d_spacetime_metric,
+    const tnsr::iaa<DataVector, Dim, frame>& d_pi,
+    const tnsr::ijaa<DataVector, Dim, frame>& d_phi) noexcept {
+  GeneralizedHarmonic::BoundaryConditions::ConstraintPreservingBjorhus<Dim>
+      bjorhus_obj{
+          GeneralizedHarmonic::BoundaryConditions::detail::
+              ConstraintPreservingBjorhusType::ConstraintPreservingPhysical};
+  bjorhus_obj.dg_time_derivative(
+      dt_spacetime_metric_correction, dt_pi_correction, dt_phi_correction,
+      face_mesh_velocity, normal_covector, normal_vector, spacetime_metric, pi,
+      phi, coords, gamma1, gamma2, lapse, shift, inverse_spacetime_metric,
+      spacetime_unit_normal_vector, spacetime_unit_normal_one_form,
+      three_index_constraint, gauge_source, spacetime_deriv_gauge_source,
+      dt_spacetime_metric, dt_pi, dt_phi, d_spacetime_metric, d_pi, d_phi);
+}
+
+template <size_t Dim>
+void test_with_random_values(const DataVector& used_for_size) noexcept {
+  pypp::check_with_random_values<1>(
+      wrap_dt_vars_corrections_ConstraintPreserving<Dim>,
+      "Evolution.Systems.GeneralizedHarmonic.BoundaryConditions.Bjorhus",
+      {"dt_spacetime_metric", "dt_pi_ConstraintPreserving",
+       "dt_phi_ConstraintPreserving"},
+      {{{0.1, 0.5}}}, used_for_size, 1.e-6);
+  pypp::check_with_random_values<1>(
+      wrap_dt_vars_corrections_ConstraintPreservingPhysical<Dim>,
+      "Evolution.Systems.GeneralizedHarmonic.BoundaryConditions.Bjorhus",
+      {"dt_spacetime_metric", "dt_pi_ConstraintPreservingPhysical",
+       "dt_phi_ConstraintPreservingPhysical"},
+      {{{0.1, 0.5}}}, used_for_size, 1.e-6);
+}
+}  // namespace
+
+// [[TimeOut, 10]]
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.GeneralizedHarmonic.BCBjorhus.Cls",
+                  "[Unit][Evolution]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{""};
+
+  //   test<1>();
```

Comment:
why is the 1d case commented out?

Reply:
Re-enabled it. There was a rarely triggered [bug](https://github.com/sxs-collaboration/spectre/pull/3172#issuecomment-864992881) that I hadn't caught yet, but I found and fixed it.